### ISSUE TITLE
Ruleset ignores level and sub-rules

### ISSUE THREAD

**Issue by Rahix** [priv-kweihmann/oelint-adv#236]
from PR #235:

1. With the following rulefile:
   ```json
   {
     "oelint.var.mandatoryvar.DESCRIPTION": "error"
   }
   ```
   I get
   ```bash
   ❯ oelint-adv --rulefile .ci/oelint-rules.json --color /path/to/recipe.bb
   Loaded rules:
   	
   
   ```
2. With
   ```json
   {
     "oelint.var.mandatoryvar": "error"
   }
   ```
   I get
   ```bash
   ❯ oelint-adv --rulefile .ci/oelint-rules.json --color /path/to/recipe.bb
   Loaded rules:
   	oelint.var.mandatoryvar
   	oelint.var.mandatoryvar.DESCRIPTION
   	oelint.var.mandatoryvar.HOMEPAGE
   	oelint.var.mandatoryvar.LICENSE
   	oelint.var.mandatoryvar.SRC_URI
   	oelint.var.mandatoryvar.SUMMARY
   /path/to/recipe.bb:1:error:oelint.var.mandatoryvar.DESCRIPTION:Variable 'DESCRIPTION' should be set
   /path/to/recipe.bb:1:error:oelint.var.mandatoryvar.HOMEPAGE:Variable 'HOMEPAGE' should be set
   ```
3. With
   ```json
   {
     "oelint.var.mandatoryvar": "warning"
   }
   ```
   I get
   ```bash
   ❯ oelint-adv --rulefile .ci/oelint-rules.json --color --nowarn /path/to/recipe.bb
   Loaded rules:
   	oelint.var.mandatoryvar
   	oelint.var.mandatoryvar.DESCRIPTION
   	oelint.var.mandatoryvar.HOMEPAGE
   	oelint.var.mandatoryvar.LICENSE
   	oelint.var.mandatoryvar.SRC_URI
   	oelint.var.mandatoryvar.SUMMARY
   /path/to/recipe.bb:1:error:oelint.var.mandatoryvar.DESCRIPTION:Variable 'DESCRIPTION' should be set
   /path/to/recipe.bb:1:error:oelint.var.mandatoryvar.HOMEPAGE:Variable 'HOMEPAGE' should be set
   ```

It looks like A) sub-rules like `oelint.var.mandatoryvar.DESCRIPTION` are ignored and B) setting a non-default lint level is also ignored.
